### PR TITLE
fix: screen size change did not cause chat page rerender #870

### PR DESCRIPTION
**Pull request by FrolYo**

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Pages/ChatPage.razor [hunk 1]
```diff
@@ -135,8 +134,13 @@
...
                 return false;
             return ChatId == other.ChatId
-                && ReferenceEquals(Chat, other.Chat)
-                && ReferenceEquals(Account, other.Account);
+                   && ReferenceEquals(Chat, other.Chat)
+                   && ReferenceEquals(Account, other.Account)
+                   && ScreenSize == other.ScreenSize;
+        }
+
+        public Model ReplaceIfSame(Model? other) {
```
Comment (alexyakunin) [inline src/dotnet/Chat.UI.Blazor/Pages/ChatPage.razor:142]: "ReplaceIfSame" - not sure how to explain, but I don't like this name... I.e. seems weird / the action itself isn't self-explanatory.

Comment (alexyakunin) [reply src/dotnet/Chat.UI.Blazor/Pages/ChatPage.razor:142]: I.e. the code like `IsIdentical(a, b) ? b : a` is much easier to understand.

Comment (alexyakunin) [reply src/dotnet/Chat.UI.Blazor/Pages/ChatPage.razor:142]: Or `a.IsIdenticalTo(b) ? b : a` - whatever you prefer. `Replace` word is quite confusing here. Replace what? With what?

Comment (alexyakunin) [reply src/dotnet/Chat.UI.Blazor/Pages/ChatPage.razor:142]: @FrolYo глянь :)

Comment (FrolYo) [reply src/dotnet/Chat.UI.Blazor/Pages/ChatPage.razor]: done

**Pull request merged**